src/coreclr/src/jit/importer.cpp

```diff
@@ -2637,12 +2621,12 @@ GenTree* Compiler::impCloneExpr(GenTree*             tree,
 
     unsigned temp = lvaGrabTemp(true DEBUGARG(reason));
 
-    // impAssignTempGen() may change tree->gtType to TYP_VOID for calls which
+    // impAssignTempGenStruct() may change tree->gtType to TYP_VOID for calls which
     // return a struct type. It also may modify the struct type to a more
     // specialized type (e.g. a SIMD type).  So we will get the type from
-    // the lclVar AFTER calling impAssignTempGen().
+    // the lclVar AFTER calling impAssignTempGenStruct().
 
-    impAssignTempGen(temp, tree, structHnd, curLevel, pAfterStmt, impCurStmtOffs);
+    impAssignTempGenStruct(temp, tree, structHnd, curLevel, nullptr, impCurStmtOffs);
```

Comment:
that was a strange piece, we were passing `Statement** pAfterStmt`, but not `BasicBlock* block` that is required when you use statements inserts (to update `block->bbStmtList->SetPrevStmt(newStmt)` if we insert after the current last statement).

Reply:
So I take it that "after statement" wasn't actully used?

Reply:
We have a few methods that accept such arguments:
(unsigned curLevel, Statement** pAfterStmt = nullptr, BasicBlock* block = nullptr), but we always use either `curLevel` (when don't have statements and use `impAppendTree`) or `pAfterStmt and block ` (when we have statements and don't need `curLevel`).

So an assert that we can add to such methods is `((pAfterStmt == nullptr && block ==nullptr) || (pAfterStmt != nullptr && block !=nullptr)`.

In that example we were passing only `pAfterStmt` without `block` and the only way how it can work is if `pAfterStmt == nullptr` there. `impCloneExpr` was called with potentially non-zero `pAfterStmt` only in one place: `impAssignStructPtr` (importer:1319 under `src->gtOper == GT_MKREFANY` condition), but from that moment it is not clear for me when `impAssignStructPtr` needs `pAfterStmt` and when it doesn't, because these calls are recursive.

Reply:
I guess having multiple, dependent parameters with default values isn't such a good idea :)